app/src/main/java/com/github/swent/echo/viewmodels/HomeScreenViewModel.kt

```diff
@@ -179,16 +187,16 @@ constructor(
                         dateFilterConditions(event)
                         // filter by scope of the event
                         &&
-                        (_filtersContainer.value.epflChecked &&
-                            event.tags.any { tag -> tag.name.lowercase() == "epfl" } ||
-                            _filtersContainer.value.sectionChecked &&
-                                event.tags.any { tag -> tag.name.lowercase() == "in" } ||
-                            _filtersContainer.value.classChecked &&
+                        ((_filtersContainer.value.epflChecked &&
+                            event.tags.any { tag -> tag.name.lowercase() == "epfl" }) ||
+                            (_filtersContainer.value.sectionChecked &&
+                                event.tags.any { tag ->
+                                    tag.name.lowercase() == section.lowercase()
+                                }) ||
+                            (_filtersContainer.value.classChecked &&
                                 event.tags.any { tag ->
-                                    tag.name.lowercase() == "ba6"
-                                }) // change when we have the userProfile (take their class and
-                        // section
-                        // as strings)
+                                    tag.name.lowercase() == semester.lowercase()
```

Comment:
If the `semester` is empty (or you could also use a null value) we shuold not filter by the semester, otherwise no events will be shown. We should do the same for `section`.
Also, consider consider using multiple `.filter {}`, e.g.
```
allEventList
    .filter {} // by tags
    .filter {} // by time
    .filter {} / / by section 
    ....
```
I think it would make the code easier to read :)